### PR TITLE
Go: model `slices` package (skipping functions that involve iterating over a function)

### PR DESCRIPTION
**Pull request by owen-mc**

### CODE REVIEW

File: go/ql/lib/ext/slices.model.yml [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+      - ["slices", "", False, "MaxFunc", "", "", "Argument[0].ArrayElement", "ReturnValue", "value", "manual"]
+      - ["slices", "", False, "Min", "", "", "Argument[0].ArrayElement", "ReturnValue", "value", "manual"]
+      - ["slices", "", False, "MinFunc", "", "", "Argument[0].ArrayElement", "ReturnValue", "value", "manual"]
+      - ["slices", "", False, "Repeat", "", "", "Argument[0].ArrayElement", "ReturnValue.ArrayElement", "value", "manual"]
+      - ["slices", "", False, "Replace", "", "", "Argument[0].ArrayElement", "ReturnValue.ArrayElement", "value", "manual"]
+      - ["slices", "", False, "Replace", "", "", "Argument[3].ArrayElement", "ReturnValue.ArrayElement", "value", "manual"]
+      # Sorted should be modeled when we have a way to model iterators
+      # SortedFunc should be modeled when we have a way to model iterators
+      # SortedStableFunc should be modeled when we have a way to model iterators
+      # Vales should be modeled when we have a way to model iterators
```
Comment (smowton) [inline go/ql/lib/ext/slices.model.yml:31]:
```suggestion
      # Values should be modeled when we have a way to model iterators
```

**Review by smowton — APPROVED** [review]

**Comment by owen-mc** [comment]
@smowton Please re-review the one letter change that you suggested 😆 . I'm fine to merge this without DCA or QA. Please say if you think I should run them.

**Review by egregius313 — APPROVED** [review]
LGTM.

I started a DCA run, personally I don't think a QA run is necessary for this.

**Comment by owen-mc** [comment]
DCA runs shows nothing of note. All the projects are very old, so they wouldn't be using this new library.

**Pull request merged**